fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -4083,6 +4087,54 @@ public List<QueryStatisticsItem.FragmentInstanceInfo> getFragmentInstanceInfos()
         return result;
     }
 
+    @Override
+    public void refreshExecStatus() {
+        List<TNetworkAddress> backendAddresses = Lists.newArrayList();
+        if (this.enablePipelineXEngine) {
+            for (Long backendId : this.beToPipelineExecCtxs.keySet()) {
+                Backend backend = idToBackend.get(backendId);
+                backendAddresses.add(new TNetworkAddress(backend.getHost(), backend.getBePort()));
+            }
+        } else {
+            for (Long backendId : this.beToExecStates.keySet()) {
+                Backend backend = idToBackend.get(backendId);
+                backendAddresses.add(new TNetworkAddress(backend.getHost(), backend.getBePort()));
+            }
+        }
+
+        for (TNetworkAddress address : backendAddresses) {
```

Comment:
If there are many BEs, call one by one is very slow.

Reply:
I will change this to parallel